snapcraft/plugins/_python/_sitecustomize.py

```diff
@@ -31,7 +31,14 @@
     snapcraft_stage_dir = os.getenv("SNAPCRAFT_STAGE")
     snapcraft_part_install = os.getenv("SNAPCRAFT_PART_INSTALL")
 
-    for d in (snap_dir, snapcraft_stage_dir, snapcraft_part_install):
+    site_directories = [snapcraft_stage_dir, snapcraft_part_install]
+    # Do not include snap_dir during builds as this will include
+    # snapcraft's in-snap site directory.
+    # snapcraftctl exports SNAPCRAFT_INTERPRETER
+    if not os.getenv("SNAPCRAFT_INTERPRETER"):
+        site_directories = [snap_dir] + site_directories
```

Comment:
could just insert(0...

Reply:
Aren't these two cases now mutually exclusive?

```
if os.getenv("SNAPCRAFT_INTERPRETER"):
  snapcraft_stage_dir = os.getenv("SNAPCRAFT_STAGE")
  snapcraft_part_install = os.getenv("SNAPCRAFT_PART_INSTALL")
  site_directories = [snapcraft_stage_dir, snapcraft_part_install]
else:
  site_directories = [snap_dir]
```
When would you have SNAPCRAFT_STAGE or SNAPCRAFT_PART_INSTALL if no SNAPCRAFT_INTERPRETER?

Reply:
doh, good input

Reply:
wrt insert, I sort of preferred the concat option for visual appeal :-P

Reply:
ah yes, visual appeal indeed :D